src/client/Microsoft.Identity.Client/Extensibility/CdtAuthenticationScheme.cs

```diff
@@ -0,0 +1,296 @@
+            string headerAndPayload = sb.ToString();
+
+            if (signPayload)
+            {
+                sb.Append('.');
+                sb.Append(Base64UrlHelpers.Encode(_cdtCryptoProvider.Sign(Encoding.UTF8.GetBytes(headerAndPayload))));
+            }
+
+            return sb.ToString();
+        }
+    }
+
+    public static class CdtClaimTypes
+    {
+        #region JSON keys for Http request
+
+        /// <summary>
+        /// Access token with response cnf
+        /// 
+        /// </summary>
+        public const string Ticket = "t";
+
+        /// <summary>
+        /// Constraints specified by the client
+        /// 
+        /// </summary>
+        public const string ConstraintsToken = "c";
+
+        /// <summary>
+        /// Constraints specified by the client
+        /// 
+        /// </summary>
+        public const string Constraints = "constraints";
+
+        /// <summary>
+        /// Non-standard claim representing a nonce that protects against replay attacks.
+        /// </summary>
+        public const string Nonce = "xms_ds_nonce ";
+
+        /// <summary>
+        /// 
+        /// </summary>
+        public const string Type = "typ";
+
+        #endregion
+    }
+
+    //TODO: Add support for ECD keys
+    public class CdtCryptoProvider
+    {
+        private readonly X509Certificate2 _cert;
+
+        public CdtCryptoProvider(X509Certificate2 cert)
+        {
+            _cert = cert ?? throw new ArgumentNullException(nameof(cert));
+
+            RSA provider = _cert.GetRSAPublicKey();
+            RSAParameters publicKeyParams = provider.ExportParameters(false);
+            CannonicalPublicKeyJwk = ComputeCanonicalJwk(publicKeyParams);
+        }
+
+        public byte[] Sign(byte[] payload)
+        {
+            using (RSA key = _cert.GetRSAPrivateKey())
+            {
+                return key.SignData(
+                    payload,
+                    HashAlgorithmName.SHA256,
+                    RSASignaturePadding.Pss);
+            }
+        }
+
+        public string CannonicalPublicKeyJwk { get; }
+
+        public string CryptographicAlgorithm { get => "PS256"; }
+
+        /// <summary>
+        /// Creates the canonical representation of the JWK.  See https://tools.ietf.org/html/rfc7638#section-3
+        /// The number of parameters as well as the lexicographic order is important, as this string will be hashed to get a thumbprint
+        /// </summary>
+        private static string ComputeCanonicalJwk(RSAParameters rsaPublicKey)
+        {
+            return $@"{{""e"":""{Base64UrlHelpers.Encode(rsaPublicKey.Exponent)}"",""kty"":""RSA"",""n"":""{Base64UrlHelpers.Encode(rsaPublicKey.Modulus)}""}}";
+        }
+    }
+
+    /// <summary>
+    /// Delagated Constraint
+    /// </summary>
+    public class ConstraintDict
+    {
+        public Dictionary<string, string> Constraints { get; set; } = new Dictionary<string, string>();
+    }
+
+    public class Constraint
```

Comment:
Please remove these 2 classes Constraint and ConstraintTarget.
They are not part of the MSAL extensibility.

and SAL need them too
-> Microsoft.Identity.Abstractions.Internal